components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -144,7 +132,86 @@ class P3Microphysics : public AtmosphereProcess
     view_2d       cld_frac_i;
     view_2d       cld_frac_r;
     view_2d       dz;
-  }; // run_local_vars
+    // Assigning local variables
+    void set_variables(const int ncol, const int npack, 
+           view_2d_const pmid_, view_2d T_atm_, view_2d_const ast_, view_2d_const zi_,
+           view_2d exner_, view_2d th_atm_, view_2d cld_frac_l_, view_2d cld_frac_i_, view_2d cld_frac_r_, view_2d dz_
+           )
+    {
+      m_ncol = ncol;
+      m_npack = npack;
+      // IN
+      pmid = pmid_;
+      T_atm = T_atm_;
+      ast = ast_;
+      zi = zi_;
+      // OUT
+      exner = exner_;
+      th_atm = th_atm_;
+      cld_frac_l = cld_frac_l_;
+      cld_frac_i = cld_frac_i_;
+      cld_frac_r = cld_frac_r_;
+      dz = dz_;
+    } // set_variables
+  }; // p3_preamble
+  /* --------------------------------------------------------------------------------------------*/
+  // Most individual processes have a post-processing step that derives variables needed by the rest
+  // of the model, using outputs from this process.
+  // Structure to handle the generation of data needed by the rest of the model based on output from
+  // p3_main.
+  struct p3_postamble {
+    p3_postamble() = default; 
+    // Functor for Kokkos loop to pre-process every run step
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const int icol) const {
+      for (int ipack=0;ipack<m_npack;ipack++) {
+        // Update the atmospheric temperature and the previous temperature.
+        const Spack oexner(exner(icol,ipack));
+        const Smask oexner_mask(!isnan(oexner) and oexner>0.0);
```

Comment:
Neither here nor there, but if you like the English word versions of the logical operators, you can also use `not` instead of `!`.